### ISSUE TITLE
Having very odd git issue with terragrunt

### ISSUE THREAD

**Issue by lanmalkieri** [gruntwork-io/terragrunt#447]
Hey all, having an extremely odd issue with terragrunt when using a git source. 

Below is the error output with DEBUG logging enabled. I do not get this error anywhere else with any other git command/scripts/interaction. Just this specific scenario, and it appears to be totally random.

Terraform version: 0.10.5
Terragrunt version: 0.14.3
Git Version: 2.16.2

Below is the error I receive. Anyone else see anything like this?
`[terragrunt] [/Users/home/git/terraform/dev@REDACTED.com=REDACTED/dev/cms] 2018/03/22 22:33:47 Running command: terraform --version
[terragrunt] 2018/03/22 22:33:47 Reading Terragrunt config file at /Users/home/git/terraform/dev@REDACTED.com=REDACTED/dev/cms/terraform.tfvars
[terragrunt] 2018/03/22 22:33:47 The --terragrunt-source-update flag is set, so deleting the temporary folder /Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY before downloading source.
[terragrunt] 2018/03/22 22:33:47 Downloading Terraform configurations from git::ssh://git@github.com/REDACTED/terraform?ref=master into /Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY using terraform init
[terragrunt] [/Users/home/git/terraform/dev@REDACTED.com=REDACTED/dev/cms] 2018/03/22 22:33:47 Initializing remote state for the s3 backend
[terragrunt] [/Users/home/git/terraform/dev@REDACTED.com=REDACTED/dev/cms] 2018/03/22 22:33:48 Running command: terraform init -backend-config=dynamodb_table=terraform-locks -backend-config=profile=REDACTED -backend-config=bucket=REDACTED-terraform-REDACTED-dev -backend-config=key=cms/terraform.tfstate -backend-config=region=us-west-2 -backend-config=encrypt=true -from-module=git::ssh://git@github.com/REDACTED/terraform?ref=master /Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY
2018/03/22 22:33:48 [INFO] Terraform version: 0.10.8  44110772d9ffd0ec3589943c6d4c93c24a5fff06
2018/03/22 22:33:48 [INFO] Go runtime version: go1.9
2018/03/22 22:33:48 [INFO] CLI args: []string{"/usr/local/Caskroom/terraform-0.10.8/0.10.8/terraform", "init", "-backend-config=dynamodb_table=terraform-locks", "-backend-config=profile=REDACTED", "-backend-config=bucket=REDACTED-terraform-REDACTED-dev", "-backend-config=key=cms/terraform.tfstate", "-backend-config=region=us-west-2", "-backend-config=encrypt=true", "-from-module=git::ssh://git@github.com/REDACTED/terraform?ref=master", "/Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY"}
2018/03/22 22:33:48 [DEBUG] Attempting to open CLI config file: /Users/home/.terraformrc
2018/03/22 22:33:48 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2018/03/22 22:33:48 [DEBUG] CLI config is &main.Config{Providers:map[string]string{}, Provisioners:map[string]string{}, DisableCheckpoint:false, DisableCheckpointSignature:false, PluginCacheDir:"~/.terraform_providers", Credentials:map[string]map[string]interface {}(nil), CredentialsHelpers:map[string]*main.ConfigCredentialsHelper(nil)}
2018/03/22 22:33:48 [INFO] CLI command args: []string{"init", "-backend-config=dynamodb_table=terraform-locks", "-backend-config=profile=REDACTED", "-backend-config=bucket=REDACTED-terraform-REDACTED-dev", "-backend-config=key=cms/terraform.tfstate", "-backend-config=region=us-west-2", "-backend-config=encrypt=true", "-from-module=git::ssh://git@github.com/REDACTED/terraform?ref=master", "/Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY"}
Copying configuration from "git::ssh://git@github.com/REDACTED/terraform?ref=master"...
2018/03/22 22:33:51 [DEBUG] plugin: waiting for all plugin processes to complete...
Error copying source module: error downloading 'ssh://git@github.com/REDACTED/terraform?ref=master': error running /usr/local/bin/git:
`

**Comment by brikis98** [comment]
> Error copying source module: error downloading 'ssh://git@github.com/REDACTED/terraform?ref=master': error running /usr/local/bin/git:

Can you `git clone` the repo?

**Comment by lanmalkieri** [comment]
Yeah git clone works fine, additionally on a subsequent run of the exact same command with no other changes, it works without issue. 

My original thought was some network weirdness or a local machine issue, but I am able to replicate this on multiple networks on multiple machines. Super odd, and I'm kind of at a loss at this point. 

Really my purpose here was to see if others were experiencing something similar to this error, I'm not convinced it's specific to terraform/terragrunt but the terragrunt command is the single only thing that I can get to generate/replicate this error. Which makes me wonder if there is some weird race condition or something going on.

**Comment by brikis98** [comment]
> Yeah git clone works fine, additionally on a subsequent run of the exact same command with no other changes, it works without issue.

That makes it even weirder...

Under the hood, Terragrunt just calls `terraform init` to check out the code. Terraform, in turn, should use your local `git` client. 

One thing I noticed is you listed:

```
Terraform version: 0.10.5
```

But I also see in your log output:

```
2018/03/22 22:33:48 [INFO] Terraform version: 0.10.8  44110772d9ffd0ec3589943c6d4c93c24a5fff06
```

How did you install Terragrunt? Did you use snap or asdf by any chance?

**Comment by lanmalkieri** [comment]
Ah right sorry. 

I wrote a bash function because I was annoyed remembering what I had in req.tf per environment/account. 

```make() {
    if [[ -f ./requirements.tf ]]; then
        version=$(grep 'required_version' ./requirements.tf |grep -o '[0-9]\.[0-9][0-9]\.[0-9]')
        chtf ${version}
        /usr/bin/make "$@"
    else
        /usr/bin/make "$@"
    fi
}```


So when I run terragrunt it changes version to whatever it needs to change. 

So for this run, yes it was 0.10.8. My mistake.

**Comment by tamsky** [comment]
Wow, small world.

Christopher, is this bug still affecting you?

Curious from this line:
```
2018/03/22 22:33:48 [INFO] CLI command args: []string{"init", "-backend-config=dynamodb_table=terraform-locks", "-backend-config=profile=REDACTED", "-backend-config=bucket=REDACTED-terraform-REDACTED-dev", "-backend-config=key=cms/terraform.tfstate", "-backend-config=region=us-west-2", "-backend-config=encrypt=true", "-from-module=git::ssh://git@github.com/REDACTED/terraform?ref=master", "/Users/home/.terragrunt/viPWZDclNlH0zcsF3yoMY038LyM/5IJW0DEIIMMzO4MOEPUco2z_5rY"}
```
Does your user actually have write access to create `/Users/home/.terragrunt/` or is that path a product of redacting the output, and your example is actually closer to `/Users/home/$USER/.terragrunt/` in practice?
